lib/_http_outgoing.js

```diff
@@ -814,6 +835,17 @@ function write_(msg, chunk, encoding, callback, fromEnd) {
     err = new ERR_STREAM_DESTROYED('write');
   }
 
+  if (msg && !msg.destroyed) {
```

Comment:
I think this would be more efficient for non-Content-Length responses:

```suggestion
  if (msg && !msg.destroyed && !msg._removedContLen && typeof msg._contentLength === 'number') {
```